File: platform/packages_mjs/env-resolver/src/env-gemini.x.ts
import { resolve, resolveInt, resolveFloat } from '@internal/env-resolve';

export interface GeminiEnv {
    apiKey: string | undefined;
    defaultModel: string;
    baseUrl: string;
    systemPrompt: string;
    defaultTemperature: number;
    defaultMaxTokens: number;
    timeoutMs: number;
}

export function resolveGeminiEnv(config?: Record<string, any>): GeminiEnv {
    return {
        apiKey: resolve(undefined, ['GEMINI_API_KEY'], config, 'apiKey', undefined),
        defaultModel: resolve(undefined, ['GEMINI_DEFAULT_MODEL'], config, 'defaultModel', 'flash'),
        baseUrl: resolve(undefined, ['GEMINI_BASE_URL'], config, 'baseUrl', 'https://generativelanguage.googleapis.com/v1beta/openai'),
        systemPrompt: resolve(undefined, ['GEMINI_SYSTEM_PROMPT'], config, 'systemPrompt', 'You are a helpful AI assistant powered by Gemini.'),
        defaultTemperature: resolveFloat(undefined, ['GEMINI_DEFAULT_TEMPERATURE'], config, 'defaultTemperature', 0.7),
        defaultMaxTokens: resolveInt(undefined, ['GEMINI_DEFAULT_MAX_TOKENS'], config, 'defaultMaxTokens', 1000),
        timeoutMs: resolveInt(undefined, ['GEMINI_TIMEOUT_MS'], config, 'timeoutMs', 60000),
    };
}

File: platform/packages_py/env_resolver/env_resolver/env_gemini_x.py
from dataclasses import dataclass
from typing import Any

from env_resolve import resolve, resolve_float, resolve_int


@dataclass(frozen=True)
class GeminiEnv:
    api_key: str | None
    default_model: str
    base_url: str
    system_prompt: str
    default_temperature: float
    default_max_tokens: int
    timeout_seconds: float


def resolve_gemini_env(config: dict[str, Any] | None = None) -> GeminiEnv:
    return GeminiEnv(
        api_key=resolve(None, ['GEMINI_API_KEY'], config, 'api_key', None),
        default_model=resolve(None, ['GEMINI_DEFAULT_MODEL'], config, 'default_model', 'flash'),
        base_url=resolve(None, ['GEMINI_BASE_URL'], config, 'base_url', 'https://generativelanguage.googleapis.com/v1beta/openai'),
        system_prompt=resolve(None, ['GEMINI_SYSTEM_PROMPT'], config, 'system_prompt', 'You are a helpful AI assistant powered by Gemini.'),
        default_temperature=resolve_float(None, ['GEMINI_DEFAULT_TEMPERATURE'], config, 'default_temperature', 0.7),
        default_max_tokens=resolve_int(None, ['GEMINI_DEFAULT_MAX_TOKENS'], config, 'default_max_tokens', 1000),
        timeout_seconds=resolve_float(None, ['GEMINI_TIMEOUT_SECONDS'], config, 'timeout_seconds', 60.0),
    )
